R/bmerge.R

```diff
@@ -160,6 +176,8 @@ bmerge = function(i, x, icols, xcols, roll, rollends, nomatch, mult, ops, verbos
     }
   }
 
+  .maybe_warn_mismatched_time_types(class(x[[xcol]]), class(i[[icol]]), x_name, i_name)
```

Comment:
moved the check later in the body -- I think it's possible to run a join where it would have thrown this warning, only to immediately then error because of mismatched 64-bit and 32-bit classes. better to only warn once the compatibility of the storage types has been confirmed.

Reply:
Should add a test about this in fact, #7123